AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,448 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+from editor_python_test_tools.utils import TestHelper as helper
+from PySide2 import QtWidgets, QtTest, QtCore
+from PySide2.QtCore import Qt
+from editor_python_test_tools.utils import Report
+import pyside_utils
+import editor_python_test_tools.hydra_editor_utils as hydra
+from editor_python_test_tools.editor_entity_utils import EditorEntity
+import editor_python_test_tools.QtPyO3DEEditor as QtPyO3DEEditor
+import azlmbr.editor as editor
+import azlmbr.math as math
+import azlmbr.bus as bus
+import azlmbr.legacy.general as general
+import azlmbr.scriptcanvas as scriptcanvas
+from consts.general import (SAVE_STRING, NAME_STRING,  WAIT_TIME_3, WAIT_FRAMES_200, ENTITY_STATES)
+from consts.scripting import (ASSET_EDITOR_UI, NODE_PALETTE_UI, NODE_PALETTE_QT, TREE_VIEW_QT, SEARCH_FRAME_QT,
+                              SEARCH_FILTER_QT,  NODE_INSPECTOR_TITLE_KEY, SCRIPT_EVENT_UI, PARAMETERS_QT,
+                              NODE_INSPECTOR_QT, NODE_INSPECTOR_UI, SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH)
+from consts.asset_editor import (EVENTS_QT, DEFAULT_SCRIPT_EVENT, SAVE_ASSET_AS)
+"""
+Editor Qt Object container for easy access
+"""
+qtpy_o3de_editor = QtPyO3DEEditor.QtPyO3DEEditor()
+
+class Tests():
+    new_event_created = ("New Script Event created", "Failed to create a new event")
+    child_event_created = ("Child Event created", "Failed to create Child Event")
+    parameter_created = ("Successfully added parameter", "Failed to add parameter")
+    parameter_removed = ("Successfully removed parameter", "Failed to remove parameter")
+
+def click_menu_option(window, option_text):
+    """
+    function for clicking an option from a Qt menu object. This function bypasses menu groups or categories. for example,
+    if you want to click the Open option from the "File" category provide "Open" as your menu text instead of "File" then "Open".
+
+    param window: the qt window object where the menu option is located
+    param option_text: the label string used in the menu option that you want to click
+
+    returns none
+    """
+    action = pyside_utils.find_child_by_pattern(window, {"text": option_text, "type": QtWidgets.QAction})
+    action.trigger()
+
+def save_script_event_file(self, file_path):
```

Comment:
Should we refactor this behavior into QtPyAssetEditor or an asset_editor_tools_qt.py?

Reply:
absolutely. I regret copying and pasting the entire scripting_tools file to make this one but I've made notes to help carefully deprecate the old file without losing anything.